src/System.Diagnostics.StackTrace/src/System/Diagnostics/StackTrace.netcore50aot.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using Internal.Diagnostics;
+
+namespace System.Diagnostics
+{
+    public sealed class StackTrace
+    {
+        private readonly IntPtr[] _ips;
+        private readonly bool _fNeedFileInfo;
+
+        public StackTrace(Exception exception, bool needFileInfo)
+        {
+            if (exception == null)
+            {
+                throw new ArgumentNullException("exception");
+            }
+            _ips = ExceptionExtensions.GetStackIPs(exception);
+            _fNeedFileInfo = needFileInfo;
+        }
+
+        public StackFrame[] GetFrames()
+        {
+            // There are special IPs mixed in to the list that aren't intended to be exposed as StackFrames
+            // Deliberately doing this without Linq to keep the dependency set down.
+            StackFrame[] tempFrames = new StackFrame[_ips.Length];
+            int countNonSpecialIPs = 0;
+            for (int i = 0; i < _ips.Length; i++)
+            {
+                if (_ips[i] != StackTraceHelper.SpecialIP.EdiSeparator)
+                {
+                    tempFrames[countNonSpecialIPs++] = new StackFrame(_ips[i], _fNeedFileInfo);
+                }
+            }
+            //for compat with CoreCLR, we always return null instead of 0 length array
+            if (countNonSpecialIPs == 0)
+            {
+                return null;
+            }
+            StackFrame[] frames = new StackFrame[countNonSpecialIPs];
```

Comment:
Yes this is possible (common even), however I believe StackTrace should be optimized for binary size/simplicity rather than throughput. The same scenarios that run through this code path also run through vastly more expensive operations such as dispatching exceptions and doing metadata/PDB lookups. If an important scenario came up where this was a hotspot we could easily revisit, but I think that is very unlikely.